app/src/main/java/org/mozilla/fenix/components/metrics/MetricsStorage.kt

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.mozilla.fenix.nimbus.FxNimbus
+import org.mozilla.fenix.utils.Settings
+
+interface MetricsStorage {
+    suspend fun shouldTrack(event: Event): Boolean
+
+    suspend fun updateSentState(event: Event)
+}
+
+internal class DefaultMetricsStorage(
+    private val settings: Settings,
+    private val checkDefaultBrowser: () -> Boolean,
+    private val withinInstallWindow: () -> Boolean,
+) : MetricsStorage {
+    /**
+     * Checks local state to see whether the [event] should be sent.
+     */
+    override suspend fun shouldTrack(event: Event): Boolean =
```

Comment:
Since we want these to be suspending for IO purposes, there is no way to await the results from `MetricsController::shouldTrack`, which is why the call is bundled into `MetricsController::track`